skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,237 @@
+            self.report_progress = lambda x: x
+
+    def fit(self, X, y=None, warm_start=False):
+        """Learn the features to select.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training vectors.
+        y : array-like of shape (n_samples,)
+            Target values.
+        warm_start : bool
+            Whether the fit should continue after having already
+            run, after increasing n_features_to_select.
+            Assumes it is called with the same X and y
+
+        Returns
+        -------
+        self : object
+        """
+        tags = self._get_tags()
+
+        if y is not None:
+            X, y = self._validate_data(
+                X,
+                y,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+                multi_output=True,
+            )
+        else:
+            X = check_array(
+                X,
+                accept_sparse="csc",
+                ensure_min_features=2,
+                force_all_finite=not tags.get("allow_nan", True),
+            )
+
+        n_features = X.shape[1]
+
+        error_msg = (
+            "n_features_to_select must be either None, an "
+            "integer in [1, n_features - 1] "
+            "representing the absolute "
+            "number of features, or a float in (0, 1] "
+            "representing a percentage of features to "
+            f"select. Got {self.n_features_to_select} features and "
+            f"an input with {n_features} features."
+        )
+
+        if self.n_features_to_select is None:
+            n_iterations = n_features // 2
+        elif isinstance(self.n_features_to_select, numbers.Integral):
+            if not 0 < self.n_features_to_select < n_features:
+                raise ValueError(error_msg)
+            n_iterations = self.n_features_to_select
+        elif isinstance(self.n_features_to_select, numbers.Real):
+            if not 0 < self.n_features_to_select <= 1:
+                raise ValueError(error_msg)
+            n_iterations = int(n_features * self.n_features_to_select)
+        else:
+            raise ValueError(error_msg)
+
+        if warm_start:
+            if self.n_selected_ == 0:
+                raise ValueError(
+                    "Cannot fit with warm_start=True without having been previously initialized"
+                )
+            self._continue_greedy_search(X, y, n_iterations)
+        else:
+            self.n_selected_ = 0
+            self._init_greedy_search(X, y, n_iterations)
+
+        n_iterations -= self.n_selected_
+
+        for n in self.report_progress(range(n_iterations)):
+
+            new_feature_idx = self._get_best_new_feature(self.scoring, X, y)
+            if new_feature_idx is not None:
+                self._update_post_selection(X, y, new_feature_idx)
+                self._postprocess()
```

Comment:
```suggestion
                self._postprocess(X, y, new_feature_idx)
```

Reply:
This change add some flexibility, which I want to use for VoronoiFPS

Reply:
What exactly do you need to do? Everything I can think of would be better suited in `self._update_post_selection()`

Reply:
In fact, all Voronoi polyhedron operations are `postprocess`. We have already selected a point, we have already added it to our list, but we need to update the polyhedron lists for future selections. I agree to use a different approach, but this one seems the most sensible to me. And after all, if we have a `postprocess` function, why not use it :).

Reply:
I think MC made a good point (and I'll need to update _greedy) that "postprocess" should be reserved for anything that has to happen _after ALL selections are made_. I'll fix now

Reply:
It's sad enough. It's just that this is the kind of logic that the Voronoi FPS lays down much worse. I'll try to figure out how to rework the existing code in that case.

Reply:
How do you mean? If it's something that needs to be done after each selection, then it's as simple as:


def _update_post_selection(self, X, y, new_feature_idx):
    super()._update_post_selection(X, y, new_feature_idx)
    ### whatever else you need to do

Reply:
It's understandable, I just wanted to logically separate the feature selection part and the Voronoi polyhedra part.  It's about the logic of the algorithm, not how to do it technically.